dbms/src/Functions/FunctionsJson.h

```diff
@@ -1514,4 +1514,464 @@ class FunctionJsonDepth : public IFunction
+            }
+            if constexpr (is_path_nullable)
+            {
+                if (null_map_path[i])
+                {
+                    SET_NULL_AND_CONTINUE
+                }
+            }
+
+            auto path_expr_container_vec = buildPathExprContainer(path_source->getWhole());
+
+            const auto & json_val = json_source->getWhole();
+            assert(json_val.size > 0);
+            JsonBinary json_binary{json_val.data[0], StringRef{&json_val.data[1], json_val.size - 1}};
+            if (json_binary.getType() != JsonBinary::TYPE_CODE_OBJECT)
+            {
+                SET_NULL_AND_CONTINUE
+            }
+
+            auto extract_json_binaries = json_binary.extract(path_expr_container_vec);
+            if (extract_json_binaries.empty() || extract_json_binaries[0].getType() != JsonBinary::TYPE_CODE_OBJECT)
+            {
+                SET_NULL_AND_CONTINUE
+            }
+
+            auto keys = extract_json_binaries[0].getKeys();
+            JsonBinary::buildKeyArrayInBuffer(keys, write_buffer);
+
+            writeChar(0, write_buffer);
+            offsets_to[i] = write_buffer.count();
+            json_source->next();
+            path_source->next();
+        }
+
+#undef SET_NULL_AND_CONTINUE
+    }
+
+    template <bool is_json_nullable>
+    void doExecuteForConstPath(
+        const std::unique_ptr<IStringSource> & json_source,
+        const NullMap & null_map_json,
+        const std::unique_ptr<IStringSource> & path_source,
+        size_t rows,
+        ColumnString::Chars_t & data_to,
+        ColumnString::Offsets & offsets_to,
+        NullMap & null_map_to) const
+    {
+        // build path expr for const path col first.
+        auto path_expr_container_vec = buildPathExprContainer(path_source->getWhole());
+
+#define SET_NULL_AND_CONTINUE             \
+    null_map_to[i] = 1;                   \
+    writeChar(0, write_buffer);           \
+    offsets_to[i] = write_buffer.count(); \
+    json_source->next();                  \
+    continue;
+
+        // reserve only for the char 0 of string end.
+        // Keys are typically small, so space is not reserved specifically for them. Instead, buffer is used to expand capacity.
+        JsonBinary::JsonBinaryWriteBuffer write_buffer(data_to, rows);
+        for (size_t i = 0; i < rows; ++i)
+        {
+            if constexpr (is_json_nullable)
+            {
+                if (null_map_json[i])
+                {
+                    SET_NULL_AND_CONTINUE
+                }
+            }
+
+            const auto & json_val = json_source->getWhole();
+            assert(json_val.size > 0);
+            JsonBinary json_binary{json_val.data[0], StringRef{&json_val.data[1], json_val.size - 1}};
+            if (json_binary.getType() != JsonBinary::TYPE_CODE_OBJECT)
+            {
+                SET_NULL_AND_CONTINUE
+            }
+
+            auto extract_json_binaries = json_binary.extract(path_expr_container_vec);
```

Comment:
Since `path` is a constant, why not extract it before the loop?

Reply:
But `json_binary` isn't constant.

Reply:
OK